core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -359,10 +361,24 @@ private void removeExpiredFiles(List<Snapshot> snapshots, Set<Long> validIds, Se
                 }
               }
             });
-    deleteDataFiles(manifestsToScan, manifestsToRevert, validIds);
-    deleteMetadataFiles(manifestsToDelete, manifestListsToDelete);
+    if (cleanExpiredFiles) {
+      deleteDataFiles(manifestsToScan, manifestsToRevert, validIds);
+      deleteMetadataFiles(manifestsToDelete, manifestListsToDelete);
+    } else {
+      Set<String> filesToDelete = findFilesToDelete(manifestsToScan, manifestsToRevert, validIds);
+      expireSnapshotResult = ExpireSnapshotResult
+          .builder()
+          .addManifestFiles(manifestsToDelete)
+          .addManifestListFiles(manifestListsToDelete)
+          .addDataFiles(filesToDelete)
+          .build();
+    }
   }
 
+  public ExpireSnapshotResult getExpiredSnapshotResult() {
```

Comment:
Iceberg doesn't use `get` in method names because it signals that either there is a better noun (like find) or it is unnecessary and should be omitted.